test/helpers.go

```
@@ -3248,6 +3248,44 @@ func GetAzureAuthDescription(mode AzureAuthMode) string {
+// If not already logged in, it attempts service principal login using
+// AZURE_CLIENT_ID, AZURE_CLIENT_SECRET, and AZURE_TENANT_ID env vars.
+// After login, sets the active subscription if AZURE_SUBSCRIPTION_ID is set.
+func EnsureAzureCliLogin(t *testing.T) error {
+	t.Helper()
+
+	if _, err := RunCommandQuiet(t, "az", "account", "show"); err == nil {
+		return nil
+	}
+
+	if !HasServicePrincipalCredentials() {
+		return fmt.Errorf("Azure CLI not logged in and no service principal credentials available (need AZURE_CLIENT_ID, AZURE_CLIENT_SECRET, AZURE_TENANT_ID)")
```

golangci-lint (2.11.4)

[error] ST1005: error strings should not be capitalized

(staticcheck)